### PR TITLE
Post delivery details screen

### LINKED ISSUE #8: The screen for selection of post office
A user should have an opportunity to select post office if they choose this delivery method.
They should pick up:
- a post company (should be hardcoded 2-3 options)
- an office of the post company (list of post offices for each company should be hardcoded; should be up to 10 items)

All fields are required

### PR DESCRIPTION
**Pull request by coding-vibe**

### CODE REVIEW

File: src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import POST_COMPANIES_OPTIONS from 'constants/postCompaniesOptions';
+import POST_OFFICES_OPTIONS from 'constants/postOfficesOptions';
+
+const { ukrposhta, novaPost, meestPoshta } = POST_OFFICES_OPTIONS;
+
+type PostDeliveryDetailsType =
+  InitialFormValuesType[FormScreens.POST_DELIVERY_DETAILS];
+
+const getPostOfficeOptions = (postCompany: PostCompanies) => {
+  switch (postCompany) {
```
Comment (coding-vibe) [inline src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx]: Use values from enum

File: src/constants/postCompaniesOptions.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import PostCompanies from 'constants/postCompanies';
+
+const { UKRPOSHTA, NOVA_POST, MEEST_POSHTA } = PostCompanies;
```
Comment (coding-vibe) [inline src/constants/postCompaniesOptions.ts]: Don't destructure me

File: src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  console.log(isChanged);
+
+  return (
+    isChanged && (
+      <FormSpy
+        onChange={(props) => {
+          // eslint-disable-next-line no-param-reassign, react/prop-types
+          props.values.postOffice = null;
+        }}
+      />
```
Comment (singvarr) [inline src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx]: Disable me if `value.postCompany` isn't selected. Add tooltip on hover for the disabled element that will explain that a user should select `postCompany` first

File: src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+} from 'contexts/WizardFormContext';
+import { validateIsRequired } from 'utils/validation';
+import PostCompanies from 'constants/postCompanies';
+import POST_COMPANIES_OPTIONS from 'constants/postCompaniesOptions';
+import POST_OFFICES_OPTIONS from 'constants/postOfficesOptions';
+
+type PostDeliveryDetailsType =
+  InitialFormValuesType[FormScreens.POST_DELIVERY_DETAILS];
+
+const getPostOfficeOptions = (postCompany: PostCompanies) => {
```
Comment (singvarr) [inline src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx]: Move me to component

File: src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      throw new Error('New post company found');
+  }
+};
+
+interface Props {
+  postCompany?: PostCompanies | null;
+}
+
+const PostOfficeSelection = forwardRef(({ postCompany }: Props, ref) => (
+  <Box sx={{ mb: 2 }}>
```
Comment (singvarr) [inline src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx]: You don't need conditional render here

File: src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  const { onSaveFormValues } = useContext(WizardFormContext);
+
+  return (
+    <Form<PostDeliveryDetailsType>
+      onSubmit={(values) => {
+        onSaveFormValues(FormScreens.POST_DELIVERY_DETAILS, values);
+      }}
+      render={({ handleSubmit, pristine, submitting, values }) => (
+        <>
+          <CustomFormSpy postCompany={values.postCompany} />
```
Comment (singvarr) [inline src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx]: Put me after `form` tag. Let's discuss today why

File: src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    default:
+      throw new Error('New post company found');
+  }
+};
+
+interface Props {
+  postCompany?: PostCompanies | null;
+}
+
+const PostOfficeSelection = forwardRef(({ postCompany }: Props, ref) => (
```
Comment (singvarr) [inline src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx]: You don't need `forwardRef` and conditional rendering here. But in order to show a tooltip message, you need to wrap contents of tooltip in html element, e.g. `span`. Check this [example](https://mui.com/material-ui/react-tooltip/#disabled-elements)

File: src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+import WizardFormContext, {
+  InitialFormValuesType,
+} from 'contexts/WizardFormContext';
+import { validateIsRequired } from 'utils/validation';
+import PostCompanies from 'constants/postCompanies';
+import POST_COMPANIES_OPTIONS from 'constants/postCompaniesOptions';
+import POST_OFFICES_OPTIONS from 'constants/postOfficesOptions';
+
+const getPostOfficeOptions = (postCompany: PostCompanies) => {
+  switch (postCompany) {
```
Comment (singvarr) [inline src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx]: Let's put all logic that defines options here. Handle a situation when there is no `postCompany` here

**Pull request merged**